app/lib/api/auth.ts

```diff
@@ -0,0 +1,66 @@
+import axios from 'axios';
+
+const getRedirectUri = () => {
+    if (!process.browser) {
+        // Avoid some nextjs compilation errors regarding window being undefined
+        return undefined;
+    }
+    return `${window.location.origin}/auth_redirect`;
+};
+
+export const generateAuthUrl = (state: string): string => {
+    const url = 'https://api.trackmania.com/oauth/authorize';
+
+    const params = {
+        response_type: 'code',
+        client_id: process.env.NEXT_PUBLIC_CLIENT_ID,
+        redirect_uri: getRedirectUri(),
+        state,
+    };
+
+    const res = axios.getUri({ url, params });
+
+    return res;
+};
+
+interface AuthorizationResponse {
+    displayName: string;
+    accountId: string;
+}
+export const authorizeWithAccessCode = async (accessCode: string): Promise<AuthorizationResponse> => {
+    const url = `${process.env.NEXT_PUBLIC_API_URL}/authorize`;
+
+    const params = {
+        code: accessCode,
+        redirect_uri: getRedirectUri(),
+    };
+
+    // TODO: use custom axios instance with default config for withCredentials
+    const { data } = await axios.post(url, params, { withCredentials: true });
+
+    return data;
+};
+
+interface MeResponse {
+    displayName: string;
+    accountId: string;
+}
+export const fetchMe = async (): Promise<MeResponse | undefined> => {
```

Comment:
I don't know, something bothers me about this function name - can we call it `fetchLoggedInUser` or something like that? Otherwise you always have to remember what "me" means in this context.